### PR TITLE
Check code style with Checkstyle

### PR DESCRIPTION
**Pull request by wb9688**
#### What is it?

- [ ] Bug fix
- [ ] Feature
- [x] Meta

#### Long description of the changes in your PR

This PR will let Travis check the code style of all PRs. Do you think this should be run before every build as well?

I'm also going to fix all the current code style issues, except in GigaCrap, so I excluded that already in `build.gradle`.

The Checkstyle configuration is based on [this](https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/sun_checks.xml), but I disabled some checks.

I'm going to implement this for NewPipeExtractor as well.

#### Agreement

- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.

### CODE REVIEW

**Comment by XiangRongLin** [comment]
How can i import this into Android Studio, so that the autoformatter will follow those rules?

**Comment by B0pol** [comment]
And another question, how to locally check if the code is good?

**Comment by Poolitzer** [comment]
> Do you think this should be run before every build as well?
Not needed if you fix everything before you merge this.

>  and added in PR template
Happily.

**Comment by connectety** [comment]
Am I correct in the assumption, that this is still a draft, because the build fails, because the formatting isn't good?
If not, what needs to be done before merging?

**Comment by B0pol** [comment]
You are right, and @wb9688 is fixing them. There was about 8000 issues, now about 3000.

**Comment by wb9688** [comment]
@B0pol and @Poolitzer: It doesn't need to be added to the PR template. It'll run on every build, and make the build fail when there are code style issues.

@connectety: That's correct and I'm fixing them.

**Comment by Poolitzer** [comment]
> @B0pol and @Poolitzer: It doesn't need to be added to the PR template. It'll run on every build, and make the build fail when there are code style issues.

The point is telling developers to run it locally so they dont push it, geta a travis fail, figure out what the reason is, and then fix it in an unnecessary commit

**Comment by wb9688** [comment]
@Poolitzer: No, as I said, it'll run on every single build. So if you press the run or build button in Android Studio (or do e.g. `./gradlew assembleDebug`), it'll fail when there are code style issues.

**Comment by XiangRongLin** [comment]
@wb9688 
> 
> 
>No, as I said, it'll run on every single build. So if you press the run or build button in Android Studio (or do e.g. `./gradlew assembleDebug`), it'll fail when there are code style issues.

I would find the build failing very annoying while developing and adding some temporary (log) statements, which may not adhere to the checkstyle.

File: app/src/main/java/org/schabi/newpipe/download/DownloadDialog.java [hunk 1]
```diff
@@ -272,21 +294,20 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
...
-            public void onProgressChanged(SeekBar seekbar, int progress, boolean fromUser) {
-                progress++;
-                prefs.edit().putInt(getString(R.string.default_download_threads), progress).apply();
-                threadsCountTextView.setText(String.valueOf(progress));
+            public void onProgressChanged(final SeekBar seekbar, final int progress,
+                                          final boolean fromUser) {
+                final int newProgress = progress + 1;
+                prefs.edit().putInt(getString(R.string.default_download_threads), newProgress)
+                        .apply();
+                threadsCountTextView.setText(String.valueOf(newProgress));
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/download/DownloadDialog.java:303]: Not sure if this is really needed

File: app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java [hunk 2]
```diff
@@ -105,110 +105,98 @@
...
     private TextView detailPositionView;
-
     private LinearLayout videoDescriptionRootLayout;
     private TextView videoUploadDateView;
     private TextView videoDescriptionView;
-
     private View uploaderRootLayout;
     private TextView uploaderTextView;
     private ImageView uploaderThumb;
-
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java:179]: I think all of these newlines should stay there

File: app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java [hunk 3]
```diff
@@ -105,110 +105,98 @@
...
     @Nullable
     private Disposable positionSubscriber = null;
-
     private List<VideoStream> sortedVideoStreams;
     private int selectedVideoStreamIndex = -1;
-
-    /*//////////////////////////////////////////////////////////////////////////
-    // Views
-    //////////////////////////////////////////////////////////////////////////*/
-
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java:147]: This comment should not be moved: the things above are not views

File: app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java [hunk 4]
```diff
@@ -363,13 +374,11 @@ protected void onRestoreInstanceState(@NonNull Bundle savedState) {
 
     }
 
-    /*//////////////////////////////////////////////////////////////////////////
-    // OnClick
-    //////////////////////////////////////////////////////////////////////////*/
-
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java:369]: Why was this moved?

File: app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java [hunk 5]
```diff
@@ -450,12 +465,8 @@ private void toggleTitleAndDescription() {
         }
     }
 
-    /*//////////////////////////////////////////////////////////////////////////
-    // Init
-    //////////////////////////////////////////////////////////////////////////*/
-
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java:456]: Why was this moved?

File: app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java [hunk 6]
```diff
@@ -544,41 +553,42 @@ protected void initListeners() {
...
+            IMAGE_LOADER.displayImage(info.getUploaderAvatarUrl(), uploaderThumb,
                     ImageDisplayConstants.DISPLAY_AVATAR_OPTIONS);
         }
     }
 
-
-    /*//////////////////////////////////////////////////////////////////////////
-    // Menu
-    //////////////////////////////////////////////////////////////////////////*/
-
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java:573]: Why was this moved?

File: app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java [hunk 7]
```diff
@@ -637,96 +647,93 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
-            public void onNothingSelected(AdapterView<?> parent) {
-            }
+            public void onNothingSelected(final AdapterView<?> parent) { }
         });
     }
 
-    /*//////////////////////////////////////////////////////////////////////////
-    // OwnStack
-    //////////////////////////////////////////////////////////////////////////*/
-
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java:677]: Why was this moved?

File: app/src/main/java/org/schabi/newpipe/fragments/list/BaseListFragment.java [hunk 8]
```diff
@@ -93,28 +102,23 @@ public void onResume() {
...
                 infoListAdapter.notifyDataSetChanged();
             }
             updateFlags = 0;
         }
     }
 
-    /*//////////////////////////////////////////////////////////////////////////
-    // State Saving
-    //////////////////////////////////////////////////////////////////////////*/
-
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/fragments/list/BaseListFragment.java:107]: Why was this moved?

File: app/src/main/java/org/schabi/newpipe/fragments/list/channel/ChannelFragment.java [hunk 9]
```diff
@@ -63,41 +63,39 @@
...
+                                              final String name) {
         ChannelFragment instance = new ChannelFragment();
         instance.setInitialData(serviceId, url, name);
         return instance;
     }
 
-    /*//////////////////////////////////////////////////////////////////////////
-    // LifeCycle
-    //////////////////////////////////////////////////////////////////////////*/
-
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/fragments/list/channel/ChannelFragment.java:98]: Why was this moved?

File: app/src/main/java/org/schabi/newpipe/fragments/list/channel/ChannelFragment.java [hunk 10]
```diff
@@ -145,21 +154,19 @@ protected View getListHeader() {
         return headerRootLayout;
     }
 
-    /*//////////////////////////////////////////////////////////////////////////
-    // Menu
-    //////////////////////////////////////////////////////////////////////////*/
-
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/fragments/list/channel/ChannelFragment.java:151]: Why was this moved?

File: app/src/main/java/org/schabi/newpipe/fragments/list/channel/ChannelFragment.java [hunk 11]
```diff
@@ -107,29 +105,40 @@ public void setUserVisibleHint(boolean isVisibleToUser) {
...
+        }
+        if (subscribeButtonMonitor != null) {
+            subscribeButtonMonitor.dispose();
+        }
     }
 
     /*//////////////////////////////////////////////////////////////////////////
-    // Init
+    // Menu
     //////////////////////////////////////////////////////////////////////////*/
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/fragments/list/channel/ChannelFragment.java:137]: Why was this moved?

File: app/src/main/java/org/schabi/newpipe/fragments/list/channel/ChannelFragment.java [hunk 12]
```diff
@@ -197,22 +208,16 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return true;
     }
 
-    /*//////////////////////////////////////////////////////////////////////////
-    // Channel Subscription
-    //////////////////////////////////////////////////////////////////////////*/
-
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/fragments/list/channel/ChannelFragment.java:203]: Why was this moved?

File: app/src/main/java/org/schabi/newpipe/fragments/list/playlist/PlaylistFragment.java [hunk 13]
```diff
@@ -276,19 +294,20 @@ public void handleResult(@NonNull final PlaylistInfo result) {
                     }
                 });
             }
-        } else { // Else : say we have no uploader
+        } else { // Else say we have no uploader
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/fragments/list/playlist/PlaylistFragment.java:297]:
```suggestion
        } else { // Otherwise say we have no uploader
```

File: app/src/main/java/org/schabi/newpipe/fragments/list/search/SearchFragment.java [hunk 14]
```diff
@@ -114,26 +113,19 @@
...
     @State
     protected boolean wasSearchFocused = false;
-
     private Map<Integer, String> menuItemToFilterName;
     private StreamingService service;
     private String currentPageUrl;
     private String nextPageUrl;
     private String contentCountry;
     private boolean isSuggestionsEnabled = true;
-
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/fragments/list/search/SearchFragment.java:131]: These spaces should stay here

File: app/src/main/java/org/schabi/newpipe/fragments/list/search/SearchFragment.java [hunk 15]
```diff
@@ -161,45 +154,49 @@ public static SearchFragment getInstance(int serviceId, String searchString) {
...
      * Set wasLoading to true so when the fragment onResume is called, the initial search is done.
      */
     private void setSearchOnResume() {
         wasLoading.set(true);
     }
 
-    /*//////////////////////////////////////////////////////////////////////////
-    // Fragment's LifeCycle
-    //////////////////////////////////////////////////////////////////////////*/
-
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/fragments/list/search/SearchFragment.java:174]: Why was this moved?

File: app/src/main/java/org/schabi/newpipe/fragments/list/search/SearchFragment.java [hunk 16]
```diff
@@ -289,29 +310,31 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
     }
 
     /*//////////////////////////////////////////////////////////////////////////
-    // Init
+    // State Saving
     //////////////////////////////////////////////////////////////////////////*/
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/fragments/list/search/SearchFragment.java:314]: Why was this moved?

File: app/src/main/java/org/schabi/newpipe/fragments/list/videos/RelatedVideosFragment.java [hunk 17]
```diff
@@ -28,68 +29,73 @@
...
+    public static RelatedVideosFragment getInstance(final StreamInfo info) {
         RelatedVideosFragment instance = new RelatedVideosFragment();
         instance.setInitialData(info);
         return instance;
     }
 
-    /*//////////////////////////////////////////////////////////////////////////
-    // LifeCycle
-    //////////////////////////////////////////////////////////////////////////*/
-
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/fragments/list/videos/RelatedVideosFragment.java:52]: Why was this moved?

File: app/src/main/java/org/schabi/newpipe/fragments/list/videos/RelatedVideosFragment.java [hunk 18]
```diff
@@ -143,62 +159,63 @@ public void handleNextItems(ListExtractor.InfoItemsPage result) {
     }
 
     /*//////////////////////////////////////////////////////////////////////////
-    // OnError
+    // Utils
     //////////////////////////////////////////////////////////////////////////*/
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/fragments/list/videos/RelatedVideosFragment.java:163]: Why was this moved?

File: app/src/main/java/org/schabi/newpipe/fragments/list/videos/RelatedVideosFragment.java [hunk 19]
```diff
@@ -143,62 +159,63 @@ public void handleNextItems(ListExtractor.InfoItemsPage result) {
...
-        showSnackBarError(exception, UserAction.REQUESTED_STREAM, NewPipe.getNameOfService(serviceId), url, R.string.general_error);
+        showSnackBarError(exception, UserAction.REQUESTED_STREAM,
+                NewPipe.getNameOfService(serviceId), url, R.string.general_error);
         return true;
     }
 
-    /*//////////////////////////////////////////////////////////////////////////
-    // Utils
-    //////////////////////////////////////////////////////////////////////////*/
-
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/fragments/list/videos/RelatedVideosFragment.java:161]: Why was this moved?

**Comment by wb9688** [comment]
@Stypox: Those moves are probably caused by me using the Android Studio's reformatter initially on all files, and/or Git fucking up rebase.

**Comment by wb9688** [comment]
@TobiGr: I fixed everything that was pointed out by @Stypox (also in the other files he didn't check), so you could merge it now.

**Comment by B0pol** [comment]
I picked your branch, ran `runCheckStyle`
> Checkstyle files with violations: 55
> Checkstyle violations by severity: [warning:161]

**Comment by wb9688** [comment]
@B0pol: Yes, it's intended that there are some warnings, but those are not errors. We should maybe disable `VisibilityModifier` though, or fix those warnings in another PR.

**Comment by B0pol** [comment]
I removed my message because I had no more this after rerun, but it was just a bug
Current state:
> Checkstyle files with violations: 55
> Checkstyle violations by severity: [warning:161]

> @B0pol: Yes, it's intended that there are some warnings, but those are not errors. We should maybe disable `VisibilityModifier` though, or fix those warnings in another PR.

Why another pr? It's harder to find the error if there are 161 warnings, if you remove these warnings, gradle will show up only the line with the error.

also, why is newpipe.streams excluded?

**Comment by wb9688** [comment]
>Why another pr? It's harder to find the error if there are 161 warnings, if you remove these warnings, gradle will show up only the line with the error.

Because it's a lot of work and not really needed. We could just disable `VisibilityModifier` in this PR though.

>also, why is newpipe.streams excluded?

I didn't want to do that, since I had `VisibilityModifier` as `error` initially, and there it would be a lot of even more pointless work. However now that it's a warning or even disable, I could do that.

File: checkstyle.xml [hunk 20]
```diff
@@ -0,0 +1,184 @@
...
+        <module name="LeftCurly"/>
+        <module name="NeedBraces"/>
+        <module name="RightCurly"/>
+
+        <!-- Checks for common coding problems               -->
+        <!-- See https://checkstyle.org/config_coding.html -->
+        <module name="EmptyStatement"/>
+        <module name="EqualsHashCode">
+            <property name="severity" value="warning"/>
+        </module>
```
Comment (TobiGr) [inline checkstyle.xml:141]: Do we really need this?

File: checkstyle.xml [hunk 21]
```diff
@@ -0,0 +1,184 @@
...
+        </module>-->
+
+        <!-- Miscellaneous other checks.                   -->
+        <!-- See https://checkstyle.org/config_misc.html -->
+        <module name="ArrayTypeStyle"/>
+        <module name="FinalParameters"/>
+        <module name="TodoComment">
+            <property name="format" value="(TODO:|FIXME:)"/>
+            <property name="severity" value="warning"/>
+        </module>
```
Comment (TobiGr) [inline checkstyle.xml:174]: I think this is not needed as well.

File: checkstyle.xml (hunk 20 above)
Comment (wb9688) [reply checkstyle.xml:141]: That was in the config I mentioned, though I changed its severity to warning instead of error. Quote from Checkstyle's docs:

> Rationale: The contract of equals() and hashCode() requires that equal objects have the same hashCode. Therefore, whenever you override equals() you must override hashCode() to ensure that your class can be used in hash-based collections.

File: checkstyle.xml (hunk 21 above)
Comment (wb9688) [reply checkstyle.xml:174]: This was also in the config I mentioned, though I changed it to also pick up FIXME comments and changed its severity to warning as well. Maybe we should indeed just disable this.

File: checkstyle.xml (hunk 20 above)
Comment (Stypox) [reply checkstyle.xml:141]: I think this should stay here, as a warning, since it could help preventing nasty hash bugs when putting custom classes built in lib containers like HashMap

File: checkstyle.xml (hunk 21 above)
Comment (Stypox) [reply checkstyle.xml:174]: These are not really something to "solve". Yes, they need to be solved sooner or later, but it's not something someone creating a normal PR should care and be notified about imho.

File: app/src/main/java/org/schabi/newpipe/info_list/holder/ChannelInfoItemHolder.java [hunk 22]
```diff
@@ -31,19 +31,23 @@
...
+                               final HistoryRecordManager historyRecordManager) {
         super.updateFromItem(infoItem, historyRecordManager);
 
-        if (!(infoItem instanceof ChannelInfoItem)) return;
-        final ChannelInfoItem item = (ChannelInfoItem) infoItem;
+        if (!(infoItem instanceof ChannelInfoItem)) {
+            return;
+        }
+        final ChannelInfoItem item;
+        item = (ChannelInfoItem) infoItem;
```
[the remark this answers is not in the document]
Comment (wb9688) [reply app/src/main/java/org/schabi/newpipe/info_list/holder/ChannelInfoItemHolder.java:50]: Yes, that's stupid indeed.

File: app/src/main/java/org/schabi/newpipe/local/subscription/services/BaseImportExportService.java [hunk 23]
```diff
@@ -179,18 +192,14 @@ protected void postErrorResult(String title, String text) {
                 .setContentTitle(getString(getTitle()));
     }
 
-    /*//////////////////////////////////////////////////////////////////////////
-    // Toast
-    //////////////////////////////////////////////////////////////////////////*/
-
```
[the remark this answers is not in the document]
Comment (wb9688) [reply app/src/main/java/org/schabi/newpipe/local/subscription/services/BaseImportExportService.java:185]: I didn't see that.

File: app/src/main/java/org/schabi/newpipe/player/MainVideoPlayer.java [hunk 24]
```diff
@@ -472,49 +501,49 @@ public void onPlaybackParameterChanged(float playbackTempo, float playbackPitch,
...
-
-            this.moreOptionsButton = rootView.findViewById(R.id.moreOptionsButton);
-            this.secondaryControls = rootView.findViewById(R.id.secondaryControls);
-            this.kodiButton = rootView.findViewById(R.id.kodi);
-            this.shareButton = rootView.findViewById(R.id.share);
-            this.toggleOrientationButton = rootView.findViewById(R.id.toggleOrientation);
-            this.switchBackgroundButton = rootView.findViewById(R.id.switchBackground);
-            this.muteButton = rootView.findViewById(R.id.switchMute);
-            this.switchPopupButton = rootView.findViewById(R.id.switchPopup);
+        public void initViews(final View view) {
```
[the remark this answers is not in the document]
Comment (wb9688) [reply app/src/main/java/org/schabi/newpipe/player/MainVideoPlayer.java:504]: That was intentional, because of the `HiddenField` rule.

File: app/src/main/java/org/schabi/newpipe/player/playback/MediaSourceManager.java [hunk 25]
```diff
@@ -407,17 +459,21 @@ private void onMediaSourceReceived(@NonNull final PlayQueueItem item,
...
+     *
+     * @param item {@link PlayQueueItem} to check
+     * @return whether a correction is needed
+     */
     private boolean isCorrectionNeeded(@NonNull final PlayQueueItem item) {
         final int index = playQueue.indexOf(item);
         final ManagedMediaSource mediaSource = playlist.get(index);
         return mediaSource != null && mediaSource.shouldBeReplacedWith(item,
-                /*mightBeInProgress=*/index != playQueue.getIndex());
+                index != playQueue.getIndex());
```
[the remark this answers is not in the document]
Comment (wb9688) [reply app/src/main/java/org/schabi/newpipe/player/playback/MediaSourceManager.java:476]: I removed that because Android Studio already shows that, so it isn't needed.

[the remark this answers is not in the document]
Comment (Stypox) [reply app/src/main/java/org/schabi/newpipe/player/playback/MediaSourceManager.java:476]: Oh ok

File: app/src/main/java/org/schabi/newpipe/player/MainVideoPlayer.java (hunk 24 above)
[the remark this answers is not in the document]
Comment (Stypox) [reply app/src/main/java/org/schabi/newpipe/player/MainVideoPlayer.java:504]: Ah, I see

**Comment by TobiGr** [comment]
Happy rebasing to everyone who has a PR open ;)

**Comment by XiangRongLin** [comment]
@TobiGr 
I can't build the app because i get Errors in Mp4FromDashWriter

**Comment by wb9688** [comment]
@XiangRongLin: Did you change anything in that file? If so, update the line numbers in checkstyle-suppressions.xml.

**Comment by TobiGr** [comment]
@XiangRongLin That's strange because the build for the merge commit https://github.com/TeamNewPipe/NewPipe/commit/2403184845a762d308bf991daa9cc57111398775 succeded

**Comment by XiangRongLin** [comment]
@TobiGr 
It say there are lines exceeding 100 chars like https://github.com/TeamNewPipe/NewPipe/blob/dev/app/src/main/java/org/schabi/newpipe/streams/Mp4FromDashWriter.java#L713 or line 725, 739, 763

**Comment by wb9688** [comment]
>I'm on the dev branch without any changes. If i increase the line limit, it builds

@XiangRongLin: That's not possible if it fails on the lines you mention. E.g. line 725 definitely isn't longer than 100 characters.

@kapodamy: Would you mind fixing [the errors I've suppressed in `checkstyle-suppressions.xml`](https://github.com/TeamNewPipe/NewPipe/blob/dev/checkstyle-suppressions.xml#L14) anyway? As I said above:

>I also didn't know what formatting would be appropriate for certain arrays in there, or whether it was safe to just make some arguments `final`.

**Comment by XiangRongLin** [comment]
@wb9688 
Ah wait, I fixed the error in 713 by moving it into a new line, so the rest of the errors moved one line down. It should be 724, 738, 762

**Comment by wb9688** [comment]
@XiangRongLin: Have you set something else in `org.checkstyle.sun.suppressionfilter.config`? Try changing [this](https://github.com/TeamNewPipe/NewPipe/blob/dev/checkstyle.xml#L25-L26) to:

```xml
        <property name="file" value="checkstyle-suppressions.xml"/>
```

It should just work on `dev`. At least it does on my PC, and Travis confirms that. You could also try cloning NewPipe again.

**Comment by XiangRongLin** [comment]
@wb9688 
The only error i have is in line 713, the line is 101 character long.

The reason why the other lines also appeared is because i fixed line 713 by splitting it into two lines. This causes everything below to be moved one line down, which in turn cause the errors to be in one line lower. Now they don't match the suppression and appear as errors.
